File: backend/app/Models/Achievement.ts
import { DateTime } from 'luxon'
import { BaseModel, BelongsTo, belongsTo, column } from '@ioc:Adonis/Lucid/Orm'
import User from './User'

export default class Achievement extends BaseModel {
  @column({ isPrimary: true })
  public id: number

  @column()
  public user_id: number

  @column()
  public institution: string

  @column()
  public name: string

  @column()
  public position: string

  @column()
  public type: string

  @column({
    columnName: 'customized_type'
  })
  public customizedType: string

  @column.dateTime({ autoCreate: true })
  public createdAt: DateTime

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  public updatedAt: DateTime
  
  @belongsTo(() => User, {
    localKey: "user_id"
  })
  public user: BelongsTo<typeof User>
}

File: frontend/lib/models/achievement.dart
class Achievement {
  final String institution, name, position, type, customizedType;
  int id;

  Achievement(this.institution, this.name, this.position, this.type, this.customizedType);

  Achievement.fromJson(Map<String, dynamic> json)
      : institution = json['institution'],
        name = json['name'],
        position = json['position'],
        type = json['type'],
        customizedType = json['customized_type'],
        id = json['id'];

  Map<String, dynamic> toJson() => {
        'position': position,
        'name': name,
        'institution': institution,
        'type': type,
        'customized-type': customizedType
      };

      @override
  String toString() {
    return 'Achievement: {id: $id, name: $name, position: $position, institution: $institution, customizedType: $customizedType, type: $type}';
  }
}
